### ISSUE TITLE
End of support for various R spatial packages.

### ISSUE THREAD

**Issue by gareth-d-ga** [GeoscienceAustralia/ptha#9]
A number of R's spatial packages will end support at the end of 2023, [see here](https://stackoverflow.com/questions/69589608/what-should-i-actually-be-doing-in-response-to-this-warning-please-note-that-r).  

These are heavily used in `rptha` and associated codes. In applications they are often automatically loaded when one uses `library(rptha)`. 

Some notes that might help guide the code migration are here:
+ https://github.com/r-spatial/sf/wiki/Migrating

**Comment by gareth-d-ga** [comment]
See also [here](https://stat.ethz.ch/pipermail/r-sig-geo/2021-September/028760.html), which seems to suggest that support for `sp` may continue.

**Comment by gareth-d-ga** [comment]
Another relevant link here: https://r-spatial.org/r/2022/04/12/evolution.html

**Comment by gareth-d-ga** [comment]
The script below can help find locations that need fixing:
```r
#
# To help remove end-of-life packages, this script can scan *.R files in a folder, and grep for names of functions in a package
#

#library(rgdal)
#target_functions = c('rgdal', ls('package:rgdal'))
# Deliberately skip some that have common names
#target_functions = setdiff(target_functions, c('initialize', 'summary'))

#library(rgeos)
#target_functions = c('rgeos', ls('package:rgeos'))
#target_functions = setdiff(target_functions, c('plot', 'row.names', 'show'))

library(maptools)
target_functions = c('maptools', ls('package:maptools'), c('wrld_simpl'))
target_functions = setdiff(target_functions, c('label'))


# Find files in all directories of interest
slash_star<-function(n) paste(rep('/*', n), collapse="")
file_grep = paste0('./ptha', unlist(sapply(1:20, slash_star)), '.R')  # Upper limit deeper than all files
all_files = lapply(file_grep, Sys.glob)
stopifnot(length(all_files[[length(all_files)]]) == 0) # Ensure upper limit did not match files (so likely deep enough)

grep_all_files<-function(target_functions, all_files){

    file_matches = vector(mode='list', length=length(all_files))
    names(file_matches) = all_files

    for(filename in all_files){
        my_file = readLines(filename)

        function_matches = vector(mode='list', lengt=length(target_functions))
        names(function_matches) = target_functions

        for(target_fun in target_functions){
            function_matches[[target_fun]] = grep(target_fun, my_file)
        }

        file_matches[[filename]] = unlist(function_matches)
    }

    return(unlist(file_matches))
}

# x = grep_all_files(target_functions, all_files[[2]])

all_matches = lapply(all_files, function(x) grep_all_files(target_functions, x))
```

**Comment by gareth-d-ga** [comment]
More updates here: https://www.r-bloggers.com/2022/12/progress-on-r-spatial-evolution-dec-2022/

**Comment by gareth-d-ga** [comment]
I think I've removed the requirement for `maptools` [here](https://github.com/GeoscienceAustralia/ptha/pull/18) and for rgdal [here](https://github.com/GeoscienceAustralia/ptha/pull/19).

In the case of `rgdal`, we might find that some other packages still depend on it -- in principle they may break in future, let's see. 

Next step to remove `rgeos`.

**Comment by gareth-d-ga** [comment]
Pull request that removes rgeos: https://github.com/GeoscienceAustralia/ptha/pull/20

**Comment by gareth-d-ga** [comment]
When using the branch in pull request #20, all tests pass using `_SP_EVOLUTION_STATUS_=2 sudo R CMD check rptha_0.0.142.tar.gz`

Here the environmental variable is supposed to prevent sp from calling rgdal/rgeos internally.

**Comment by gareth-d-ga** [comment]
This should be resolved now by the merger of pull request #20
